domain/consensus/datastructures/blockstore/blockstore.go

```diff
@@ -9,85 +9,119 @@ import (
 )
 
 var bucket = dbkeys.MakeBucket([]byte("blocks"))
+var countKey = dbkeys.MakeBucket().Key([]byte("blocks-count"))
 
 // blockStore represents a store of blocks
 type blockStore struct {
 	staging  map[externalapi.DomainHash]*externalapi.DomainBlock
 	toDelete map[externalapi.DomainHash]struct{}
+	count    uint64
 }
 
 // New instantiates a new BlockStore
-func New() model.BlockStore {
-	return &blockStore{
+func New(dbContext model.DBReader) (model.BlockStore, error) {
+	blockStore := &blockStore{
 		staging:  make(map[externalapi.DomainHash]*externalapi.DomainBlock),
 		toDelete: make(map[externalapi.DomainHash]struct{}),
 	}
+
+	err := blockStore.initializeCount(dbContext)
+	if err != nil {
+		return nil, err
+	}
+
+	return blockStore, nil
+}
+
+func (bs *blockStore) initializeCount(dbContext model.DBReader) error {
+	var count uint64
```

Comment:
I'd rather if you initialize count to be 0, and not rely on it being the default (same for blockHeaderStore)